src/BuildIt.Forms/BuildIt.Forms.Controls/CameraPreviewControl.xaml.cs

```diff
@@ -99,5 +109,14 @@ public IReadOnlyList<CameraFocusMode> RetrieveSupportedFocusModes()
         {
             return RetrieveSupportedFocusModesFunc?.Invoke();
         }
+
+        /// <summary>
+        /// Retrieves the available camera facings
+        /// </summary>
+        /// <returns>The available camera facings</returns>
+        public Task<IReadOnlyList<ICamera>> RetrieveCamerasAsync()
+        {
+            return RetrieveCamerasFunc?.Invoke();
```

Comment:
if RetrieveCamerasFunc is null, this will return null - which is bad as it will break upstream code. Think you need to return a completed task if RetrieveCamerasFunc is null

Reply:
updated to handle nulls now